lib/github_heroku_deployer/git.rb

```diff
@@ -12,18 +12,17 @@ def initialize(options)
       @repo_dir = options[:repo_dir]
     end
 
-    def push_app_to_heroku(remote="heroku", branch="update-gemfilelock", &block)
+    def push_app_to_heroku(remote="heroku", branch="master", &block)
       wrapper = ssh_wrapper
       run "cd #{repo.dir}; git remote rm #{remote}" if repo.remote(remote).url
       repo.add_remote(remote, @heroku_repo)
       yield(repo) if block_given?
       @logger.info "deploying #{repo.dir} to #{repo.remote(remote).url} from branch #{branch}"
-      run "cd #{repo.dir}; git checkout #{branch}; env #{wrapper.git_ssh} git push -f #{remote} #{branch}:master"
+      run "cd #{repo.dir}; env #{wrapper.git_ssh} git push -f #{remote} #{branch}"
```

Comment:
Might as well keep the `branch:as-branch` syntax, as this method apparently supports branches other than master. (wont hurt to push master:as-master), and It will fail unless this syntax is used for any branch other than master (as you know)